src/options.rs

```diff
@@ -93,7 +93,31 @@ where
                       https://github.com/rust-lang-nursery/rust-bindgen/issues/426"),
             Arg::with_name("no-recursive-whitelist")
                 .long("no-recursive-whitelist")
-                .help("Avoid whitelisting types recursively."),
+                .help(r#"Disable whitelisting types recursively.
+
+Given that we have explicitly whitelisted the "initiate_dance_party" function in
```

Comment:
I'd avoid duplicating all the documentation here. What about just linking to `docs.rs` and mentioning that there are caveats?